lib/KOptionalText.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+
+  <span v-if="!textEmpty" :style="textStyle">
+    <template v-if="text">
+      {{ text }}
+    </template>
+    <template v-else>
+      <slot></slot>
+    </template>
+  </span>
+
+  <KEmptyPlaceholder v-else />
+
+</template>
+
+
+<script>
+
+  import KEmptyPlaceholder from './KEmptyPlaceholder.vue';
+
+  export default {
+    name: 'KOptionalText',
+    components: {
+      KEmptyPlaceholder,
+    },
+    props: {
+      /**
+       * Text to display
+       */
+      text: {
+        type: String,
+        default: null,
+      },
+      /**
+       * If provided, sets the styles of the text
+       */
+      textStyle: {
```

Comment:
for consistency, its probably better to use the `appearanceOverrides`, please see the [`KButton`](https://design-system.learningequality.org/kbutton/) as a usecase